EventFilter/CTPPSRawToDigi/src/RPixErrorChecker.cc

```diff
@@ -73,57 +93,111 @@ bool RPixErrorChecker::checkTrailer(bool& errorsInEvent, int fedId, unsigned int
+bool RPixErrorChecker::checkROC(bool& errorsInEvent, int fedId, uint32_t iD, Word32& errorWord,  Errors& errors) const
 {
   int errorType = (errorWord >> ROC_shift) & ERROR_mask;
   if likely(errorType<25) return true;
 
   switch (errorType) {
   case(25) : {
-    LogDebug("")<<"  invalid ROC=25 found (errorType=25)";
+    LogError("")<<"  invalid ROC=25 found (errorType=25)";
     errorsInEvent = true;
     break;
   }
   case(26) : {
-  //LogDebug("")<<"  gap word found (errorType=26)";
+    LogError("")<<"  gap word found (errorType=26)";
     return false;
   }
   case(27) : {
-  //LogDebug("")<<"  dummy word found (errorType=27)";
+    LogError("")<<"  dummy word found (errorType=27)";
     return false;
   }
   case(28) : {
-    LogDebug("")<<"  error fifo nearly full (errorType=28)";
+    LogError("")<<"  error fifo nearly full (errorType=28)";
     errorsInEvent = true;
     break;
   }
   case(29) : {
-    LogDebug("")<<"  timeout on a channel (errorType=29)";
+    LogError("")<<"  timeout on a channel (errorType=29)";
     errorsInEvent = true;
     if ((errorWord >> OMIT_ERR_shift) & OMIT_ERR_mask) {
-      LogDebug("")<<"  ...first errorType=29 error, this gets masked out";
+      LogError("")<<"  ...first errorType=29 error, this gets masked out";
       return false;
     }
     break;
   }
   case(30) : {
-    LogDebug("")<<"  TBM error trailer (errorType=30)";
+    LogError("")<<"  TBM error trailer (errorType=30)";
     errorsInEvent = true;
     break;
   }
   case(31) : {
-    LogDebug("")<<"  event number error (errorType=31)";
+    LogError("")<<"  event number error (errorType=31)";
     errorsInEvent = true;
     break;
   }
   default: return true;
   };
 
+ if(includeErrors) {
+   // check to see if overflow error for type 30, change type to 40 if so
+   if(errorType==30) {
+     uint32_t stateMach_bits      = 4;
+     uint32_t stateMach_shift     = 8;
+     uint32_t stateMach_mask = ~(~uint32_t(0) << stateMach_bits);
+     uint32_t stateMach = (errorWord >> stateMach_shift) & stateMach_mask;
+     if( stateMach==4 || stateMach==9 ) errorType = 40;
+   }
+
+   // store error
+   CTPPSPixelDataError error(errorWord, errorType, fedId);
+
+   errors[iD].push_back(error);
+}
+
   return false;
 }
 
+void RPixErrorChecker::conversionError(int fedId, uint32_t iD, int status, Word32& errorWord, Errors& errors) const
+{
+  switch (status) {
+  case(1) : {
+    LogError("ErrorChecker::conversionError") << " Fed: " << fedId << "  invalid channel Id (errorType=35)";
+    if(includeErrors) {
+      int errorType = 35;
+      CTPPSPixelDataError error(errorWord, errorType, fedId);
+      errors[iD].push_back(error);
+    }
+    break;
+  }
+  case(2) : {
+    LogError("ErrorChecker::conversionError")<< " Fed: " << fedId << "  invalid ROC Id (errorType=36)";
+    if(includeErrors) {
+      int errorType = 36;
+      CTPPSPixelDataError error(errorWord, errorType, fedId);
+      errors[iD].push_back(error);
+    }
```

Comment:
this ```if(includeErrors)``` piece is copy-pasted multiple times.
Please refactor to have it only once (RPixErrorChecker::checkROC is a good example)

Reply:
done